primitives/runtime/src/traits.rs

```diff
@@ -2414,14 +2428,9 @@ mod tests {
 	}
 
 	#[test]
-	fn ecdsa_verify_works() {
-		let msg = &b"test-message"[..];
-		let (pair, _) = ecdsa::Pair::generate();
-
-		let signature = pair.sign(&msg);
-		assert!(ecdsa::Pair::verify(&signature, msg, &pair.public()));
-
-		assert!(signature.verify(msg, &pair.public()));
-		assert!(signature.verify(msg, &pair.public()));
+	fn signature_verify_works() {
+		signature_verify_test!(ed25519);
+		signature_verify_test!(sr25519);
+		signature_verify_test!(ecdsa);
```

Comment:
Usage of a macro here is cool, but not sure if it's really necessary? How inconvenient is it to just write this inline? 

I tend to always err on the side of simpler code though, node-sdk team have final say.

Reply:
Not a final say, but it is fine for me. On one hand macro helps to DRY, on the other  potential failure message may be a little cryptic.

Reply:
Would be nice to make three separate tests here, especially since we have the macro, e.g. something like this maybe:

```rust
#[test]
fn ed25519_verify_works() {
    signature_verify_test!(ed25519);
}

#[test]
fn sr25519_verify_works() {
    signature_verify_test!(sr25519);
}

#[test]
fn ecdsa_verify_works() {
    signature_verify_test!(ecdsa);
}
```

Besides this personally I'm fine with the macro; it's not going anything complex anyway.

Reply:
Not a big deal but maybe you can also add

```
#[cfg(feature = "bls-experimental")]
fn bls377_verify_works() {
     signature_verify_test!(bls377)
}

#[cfg(feature = "bls-experimental")]
fn bls381_verify_works() {
     signature_verify_test!(bls381)
}
```

Reply:
Thank you for the review!
I reflected all your suggestion.
I would appreciate it if you could check.